CHANGES.rst

```diff
@@ -67,6 +67,8 @@ Unreleased
 
 * ``prawcore.BadRequest`` should no longer be raised. Instead a more useful
   :class:`.RedditAPIException` instance will be raised.
+* Set the default comment sort to ``confidence`` instead of ``best`` because it
+  turns out ``best`` isn't actually the correct value for the parameter.
```

Comment:
In that case, what is actually being returned?

Reply:
Whatever the user's default is, which, unless they changed it, happens to be "confidence".

Reply:
I see. Actually, why shouldn't we use the user's default? If I wanted comments to be sorted by top, it should be sorted by top unless the moderators set a sort method.

Reply:
That's a good question. The reason is supposed to be for consistency. PRAW behavior should be consistent across users, regardless of what their "defaults" are except for some obvious things like the subreddits on their frontpage which is a core feature of Reddit. It's unfortunate that this bug existed for so long (since PRAW4 was released).

Reply:
But for clients using PRAW as a wrapper, shouldn't it use the user settings? Furthermore, is there a way to retrieve the user's default sort?

Reply:
If they want to use their settings, then they can set the comment sort to None -- that should be an explicit decision. We could add that to the docstring.

Reply:
Would you not have to do that for every submission? It should be a Reddit setting that all submissions inherit from.

Reply:
That's a good point. As it is now, it would have to be applied for each submission. Feel free to open a separate PR to make the default comment sort a setting, as long as it continues to be overwritable on an individual submission as-is.

Reply:
It should be, because it could be implemented as an attribute `reddit.submission_sort`.

Reply:
I think `reddit.config.submission_sort` is where such a setting would live.

Reply:
I was looking at it, and I agree. I'll make a PR to introduce this and move the `validate_on_submit` system that exists.